### PR TITLE
Improvements for Admin::Budget::Investment filters

### LINKED ISSUE #1624: Improve investment's filters
Background
===

The admin interface to filter investments has grown organically and it's a little confusing to find what you are looking for. 

What we need
===
 
Reorganize the available filters to find any combination of investments.

Show filters by (currently working):
- **Administrator**
- **Valuator**
- **Heading**
- **Tags**

Adds an **"Advanced filters"** link with the following status:
- **Feasible**: _finished evaluation_ with feasibility = feasible.
- **Visible to valuators**: _in evaluation_ with active checkbox of "Show to valuator"
- **Selected**: those with active checkbox "Selected" (with selected == true)
- **Undecided**
- **Unfeasible**
- **Max supports per heading**: this filter is already done on [Madrid's fork](https://github.com/AyuntamientoMadrid/consul/blob/master/app/models/budget/investment.rb#L102)

And tabs with these **status**:
- **All**: all investment projects.
- **Without assigned admin**: projects without an assigned admin.
- **Without assigned valuator**: projects without an assigned valuator.
- **Under valuation**: admin and valuator assigned.
- **Valuation finished**: with active checkbox "Valuation finished"

Also, include this **changes**:

- Change table heading "Selection" to "Selected"

To consider:

- The tags should be "reset" to any budget (check this is working correctly)
- The investments should be ordered by supports and created at (last first) when there are still no supports.

🗒 On each of this tab/status, we have to include a line of help text to explain the current filter selected. 

Screenshot
===
advanced_filters

### PR DESCRIPTION
**Pull request by aitbw**
Where
=====
* **Related issues:** #1624, #2336, #2346 (This PR closes #1624 once merged into master)
* **Related PR's:** #2342

What
====
* Simplify the UI/UX of the `Admin::Budget::Investment` index view when the user wants to filter shown records

How
===
* Added `feasible`, `selected`, `undecided`, `unfeasible` and `max_supports_per_heading` filters
* Removed filters based on mockup found on #1624
* Added `without_valuator` and `under_valuation` tabbed filters
* Added missing `tbody` HTML tag in `investments` partial
* Replaced `selection` header on `investments` partial with `selected`
* All `Admin::Budget::Investment` filter-related failing specs were adapted to fit the new UI
* Added spec for `max_supports_per_heading` filter (Taken from [Ayuntamiento de Madrid](https://github.com/AyuntamientoMadrid/consul/blob/master/spec/features/admin/budget_investments_spec.rb#L295))
* Added partial to render a `currently applied filters` message to the user

Screenshots
====
peek 2018-01-22 08-48

Test
====
* Increased coverage for the `max_supports_per_heading` filter, fixed coverage as necessary

Warnings
========
* ~[These](https://github.com/consul/consul/blob/master/spec/features/admin/budget_investments_spec.rb#L602) [specs](https://github.com/consul/consul/blob/master/spec/features/admin/budgets_spec.rb#L159) were commented for this PR since #1624 (as shown on the mockup and the `Pending` section) are, well, pending. Awaiting further requirements to re-enable the `#winners` filter tab~

* ~The `filters-used` combination description is the only thing missing ATM~

### CODE REVIEW

File: app/controllers/admin/budget_investments_controller.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Admin::BudgetInvestmentsController < Admin::BaseController
   def index
     respond_to do |format|
       format.html
+      format.js { render layout: false }
```
Comment (MariaCheca) [inline app/controllers/admin/budget_investments_controller.rb]: Why is this necessary?

Comment (aitbw) [reply app/controllers/admin/budget_investments_controller.rb]: So we can make use of the advanced filters without reloading the entire page and just update what's needed —see GIF above.
